### PR TITLE
Initial commit of work

### PR DESCRIPTION
**Pull request by autumnfound**
Not final, but represents ~90% completion state

Signed-off-by: Martin Lowe <martin.lowe@eclipse-foundation.org>

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+		<dependency>
+			<groupId>com.github.scribejava</groupId>
+			<artifactId>scribejava-apis</artifactId>
+			<version>6.4.1</version>
+		</dependency>
+		<dependency>
+			<groupId>org.gitlab4j</groupId>
+			<artifactId>gitlab4j-api</artifactId>
+			<version>4.14.11</version>
+		</dependency>
```
Comment (mbarbero) [inline pom.xml]: as far as I can tell, this dependency is not used

File: src/main/java/org/eclipsefoundation/git/eca/helper/CommitHelper.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+	/**
+	 * Retrieves the email address associated with a commit message. This is done by
+	 * processing the body and parsing out the given footer knowing its format.
+	 * 
+	 * @param c the commit to retrieve the signed off by footer for
+	 * @return the email address in the Signed-off-by footer, or null if none could
+	 *         be found.
+	 */
+	public static String getSignedOffByEmail(Commit c) {
```
Comment (mbarbero) [inline src/main/java/org/eclipsefoundation/git/eca/helper/CommitHelper.java:62]: Please write unit tests for this method

File: src/main/java/org/eclipsefoundation/git/eca/helper/CommitHelper.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+public class CommitHelper {
+	private static final Pattern SIGNED_OFF_BY_FOOTER = Pattern.compile("Signed-off-by:(.*)<(.*@.*)>$");
+
+	/**
+	 * Validate the commits fields.
+	 * 
+	 * @param c commit to validate
+	 * @return true if valid, otherwise false
+	 */
+	public static boolean validateCommit(Commit c) {
```
Comment (mbarbero) [inline src/main/java/org/eclipsefoundation/git/eca/helper/CommitHelper.java:32]: Please write unit tests for this method

File: src/main/java/org/eclipsefoundation/git/eca/helper/CommitHelper.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+		if (c == null) {
+			return false;
+		}
+
+		boolean valid = true;
+		// check current commit data
+		if (c.getHash() == null || c.getParents() == null) {
```
Comment (mbarbero) [inline src/main/java/org/eclipsefoundation/git/eca/helper/CommitHelper.java]: some commits (initals) have no parent

**Review by mbarbero — CHANGES_REQUESTED** [review]

**Comment by autumnfound** [comment]
I've added a ton of unit tests, and added secret config fetch code for k8s secret stuff we will need later. Also fixed that I was missing license info in a lot of files. Did a lot of code clean up with some help from Sonar.

File: src/main/java/org/eclipsefoundation/git/eca/config/SecretConfigSource.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+			}
+
+			// read each of the lines of secret config that should be added
+			try (BufferedReader br = new BufferedReader(new FileReader(f))) {
+				Properties p = new Properties();
+				p.load(br);
+				secrets.putAll((Map) p);
+
+			} catch (IOException e) {
+				LOGGER.error("Error while reading in secrets configuration file.", e);
```
Comment (mbarbero) [inline src/main/java/org/eclipsefoundation/git/eca/config/SecretConfigSource.java:69]: Is it really a recoverable exception? IMO, you should throw a RuntimeExcepton wrapping `e`after logging

Comment (autumnfound) [reply src/main/java/org/eclipsefoundation/git/eca/config/SecretConfigSource.java:69]: This is a common class across multiple APIs, its a target for extraction to a separate project for consumption as a dependency. WDYT?

**Pull request merged**